### DISCUSSION TITLE
paste marked text into details fields

### DISCUSSION THREAD

**Discussion by volkerrichert** [jonaswinkler/paperless-ng#955] [Feature Requests]
Short description
I'll like to have a possibility to copy text into the input fields of a document. This makrs it faster to set up new documents after importing. E.g. you marked the date inside the doc and press a button next to the date field to insert this date.

Long description
Ist very foulty to edit new basic data form imported documents. You have to insert dates, title, tags, etc. It would be helpfull to have an new button ("P"aste in the screen for every input which add / replace the marked text into that field) 

![image](https://user-images.githubusercontent.com/640914/116063474-50b78e00-a685-11eb-97f0-666cc239e2ef.png)

Modern browser hat document.getSelection().toString() to get the selected text

Volker

**Comment by shamoon** [comment]
I can see why this would be useful. I started implementing it (video below) and have it working but the catch is it only works for the non-native PDF viewer, which is sort of a huge limitation. For the text fields its pretty straightforward but the date field would take a bit more work. So Im not sure if its "worth it"...

https://user-images.githubusercontent.com/4887959/116146115-fc1b0f80-a692-11eb-8c4d-994d3e6961ae.mov

**Reply by jonaswinkler** [reply]
Since we can't support the native PDF viewer, I'd not go for it.

Also, this doesn't exactly save all that much time. CTRL+C and CTRL+V is pretty straight forward here.

**Reply by volkerrichert** [reply]
WIth C&P you have to clear the field first and switch between Keyboard & Mouse. This will speed up the workflow.

When is nativ PDF viewer used? What is the problem with date? Decoding the string? Maybe moment.js can do the job

Volker

**Reply by shamoon** [reply]
Native PDF viewer is the default and if I had to guess (we dont have actual numbers) is used more commonly than the alternative viewer. And date pasting is solvable (yes parsing, cant just do `toString()`), its just more work, so point is overall the cost vs benefit for this feature isnt great...
